net/newsboat/files/patch-Makefile.diff

```diff
@@ -6,21 +6,24 @@ The target "doc/example-config" is a dependency of "install-examples" only, so
 it runs in the destroot phase, add it to "all" to avoid that. 
 --- Makefile
 +++ Makefile
-@@ -18,7 +18,7 @@
+@@ -18,9 +18,9 @@
  DEFINES+=-DGIT_HASH=\"$(GIT_HASH)\"
  endif
  
 -WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code
```

Comment:
May I ask why you strip `-Werror`? If you're getting any warnings, I'd like to fix them upstream. (Your CI doesn't show the output of the compilation process, so I can't tell what warnings you're getting.)

Reply:
To clarify why I ship with `-Werror` instead of using it only in CI: C++ is tricky, and it's hard to test with all possible combinations of libraries and compilers. [I run a beefy CI](https://cirrus-ci.com/build/4883140487413760), but it's not perfect. My hope is that by shipping with `-Werror`, I signal to users that warnings aren't okay, and if they see some, they should report them upstream for me to fix.

Reply:
`-Werror` is great.
I use it myself in my projects.
However, in the past, it has proven to be a bit of a nuisance in MacPorts.
For one thing, the vast majority of users do not look at build logs.
They just want to run `sudo port install newsboat` and be done.
MacPorts supports a wide variety of compilers (GCC, Apple Clang, vanilla Clang).
A new compiler comes out, and perfectly legal C++ causes the build to fail because of a new warning.
From the users point of view, MacPorts is just broken.
From what I can tell, [other package managers](https://svnweb.freebsd.org/ports/head/www/newsboat/Makefile?view=markup#l144) also remove `-Werror`.

If you feel strongly about it, I would be happy to modify my pull request.

There is another option.
MacPorts has a "test" feature to run any test suite the project provides.
If there are some tests that MacPorts can run after the build to see if newsboat is working correctly, we can make sure `-Werror` is part of **that** build.

Reply:
> For one thing, the vast majority of users do not look at build logs.

Oh. "Ports". I should've realized that you ship build instructions rather than binaries; sorry :) Yeah, that changes things significantly: if the users aren't invoking `make` manually, it probably doesn't make sense to dump compilation errors on them.

> If there are some tests that MacPorts can run after the build to see if newsboat is working correctly, we can make sure -Werror is part of that build.

That'd mean you have to rebuild the whole Newsboat from scratch, is that okay? But yeah, you can do this:

```
make distclean
CXXFLAGS='-Werror' make -jWHATEVER all test
```

Then run the tests:

```
cargo test
cd test && ./test
```

We run this [a slightly different way](https://github.com/newsboat/newsboat/blob/c200a6662ebc0ccf3f9ee5b41ca99e99af0ad472/.travis.yml#L75-L78) on our CI, but for you, I think it's sufficient to just do the above. If *anything* fails, you just forward it to us and we deal with it.

Reply:
Oh, regarding other package managers: well, I haven't talked to everyone just yet. Working on it :) I don't want to just roll in and randomly ask people to change their patches; I just wait around and chime in when the patches are updated, so I can discuss with a person who's doing the updates.

Thanks for the detailed responses!

Reply:
In this pull request, I added a testing variant where the port is built with `-Werror`.
Thank you for your help with improving this MacPorts port.

Reply:
> Oh. "Ports". I should've realized that you ship build instructions rather than binaries

We ship both. If the user installs MacPorts in the standard way and the licenses of the software and its dependencies allow it then we give the user a binary that we compiled on our servers. Otherwise the user's computer builds it from source using our instructions.